frontend/javascripts/oxalis/controller/camera_controller.js

```diff
@@ -171,87 +173,64 @@ type TweenState = {
-    const x2 = pos[0];
-    const y2 = pos[1];
-
-    const b2 = 1 / Math.sqrt((b1 * b1) / a1 / a1 + 1);
-    const a2 = (-b2 * b1) / a1;
-    const d2 = ((a1 / b1) * (y1 - y2) - x1 + x2) / (-a2 + (a1 * b2) / b1);
-
-    const intersect = [x2 + d2 * a2, y2 + d2 * b2];
-    const distance = Dimensions.distance([x1, y1], intersect);
-
-    // Approximation to center the view vertically
-    const yOffset = pos[2] - b[2] / 2;
-
-    // Calulate the x coordinate so that the vector from the camera to the cube's middle point is
-    // perpendicular to the vector going from (0, b[1], 0) to (b[0], 0, 0).
-
-    const squareLeft = -distance - padding;
-    const squareRight = diagonal - distance + padding;
-    const squareTop = diagonal / 2 + padding + yOffset;
-    const squareBottom = -diagonal / 2 - padding + yOffset;
-    const squareCenterX = (squareLeft + squareRight) / 2;
-    const squareCenterY = (squareTop + squareBottom) / 2;
-    const squareWidth = Math.abs(squareLeft - squareRight);
-
-    const height = squareWidth / aspectRatio;
-
-    to = {
-      dx: (b[1] / diagonal) * clippingOffsetFactor,
-      dy: (b[0] / diagonal) * clippingOffsetFactor,
-      dz: (-1 / 2) * clippingOffsetFactor,
-      upX: 0,
-      upY: 0,
-      upZ: -1,
-      l: squareCenterX - squareWidth / 2,
-      r: squareCenterX + squareWidth / 2,
-      t: squareCenterY + height / 2,
-      b: squareCenterY - height / 2,
-    };
+  // Factor to reduce the clipping offset of the z coordinate to get a better angle on the dataset.
+  const zReductionFactor = 0.7;
+  // Use width and height to keep the same zoom.
+  let width = tdCamera.right - tdCamera.left;
+  let height = tdCamera.top - tdCamera.bottom;
+
+  let position: Vector3;
+  let up: Vector3;
+  // Way to calculate the position and rotation of the camera:
+  // First, the camera is either positioned at the current center of the flycam or in the dataset center.
+  // Second, the camera is the moved backwards by a clipping factor into the wanted direction.
+  // Together with matching lookUp (up) vectors and keeping the width and height, the position and rotation updates correctly.
+  if (id === OrthoViews.TDView && (height <= 0 || width <= 0)) {
+    // This should only be the case when initializing the 3D-viewport.
+    const aspectRatio = getInputCatcherAspectRatio(state, OrthoViews.TDView);
+    const datasetCenter = voxelToNm(dataset.dataSource.scale, getDatasetCenter(dataset));
+    // The camera has not width and height which might be due to a bug or the camera has not been initialized.
+    // Thus we zoom out to show the whole dataset.
+    const paddingFactor = 1.1;
+    width =
+      Math.sqrt(
+        datasetExtent.width * datasetExtent.width + datasetExtent.height * datasetExtent.height,
+      ) * paddingFactor;
+    height = width / aspectRatio;
+    up = [0, 0, -1];
+    // For very high datasets that have a very low or hight z starting coordinate, the planes might not be visible.
```

Comment:
```suggestion
    // For very tall datasets that have a very low or high z starting coordinate, the planes might not be visible.
```

Reply:
I don't understand this part yet. To be in line with the current behavior, shouldn't the TD view be zoomed such that the whole dataset bounding box is visible, initially? Why would the planes not be visible in that case?